### ISSUE TITLE
[oplog] Querying system.replset is not the best

### ISSUE THREAD

**Issue by arunoda** [meteor/meteor#2121]
Hi, 

I just found this when I was having some issue with MongoHQ for Mongo 2.6. It is not wise to do a check on this collection. It does not reveal anything good. All the secondaries also have this collection and data in it.

So, the best option is to use `isMaster` command and use the following JSON it returned to do the check. You can check for the replicaSet and for the master.

One another things is, this command does work with a user with lower permissions too.

~~~json
{
  "setName" : "meteor",
  "setVersion" : 42581,
  "ismaster" : true,
  "secondary" : false,
  "hosts" : [
    "db-1.yyyy.com:27017",
    "db-3.yyyy.com:27017",
    "db-2.yyyy.com:27017"
  ],
  "primary" : "db-1.yyyy.com:27017",
  "me" : "db-1.yyyy.com:27017",
  "maxBsonObjectSize" : 16777216,
  "maxMessageSizeBytes" : 48000000,
  "maxWriteBatchSize" : 1000,
  "localTime" : ISODate("2014-05-06T23:53:24.297Z"),
  "maxWireVersion" : 2,
  "minWireVersion" : 0,
  "ok" : 1
}
~~~

This is the JSON for non-replica set

~~~json
{
	"ismaster" : true,
	"maxBsonObjectSize" : 16777216,
	"maxMessageSizeBytes" : 48000000,
	"maxWriteBatchSize" : 1000,
	"localTime" : ISODate("2014-05-07T00:08:35.185Z"),
	"maxWireVersion" : 2,
	"minWireVersion" : 0,
	"ok" : 1
}
~~~

This is JSON for single node Replica Set

~~~json
{
	"setName" : "meteor",
	"setVersion" : 1,
	"ismaster" : true,
	"secondary" : false,
	"hosts" : [
		"127.0.0.1:27017"
	],
	"primary" : "127.0.0.1:27017",
	"me" : "127.0.0.1:27017",
	"maxBsonObjectSize" : 16777216,
	"maxMessageSizeBytes" : 48000000,
	"maxWriteBatchSize" : 1000,
	"localTime" : ISODate("2014-05-07T00:09:41.560Z"),
	"maxWireVersion" : 2,
	"minWireVersion" : 0,
	"ok" : 1
}
~~~

**Comment by glasser** [comment]
This is fixed in d584f08 on the mongo-2.6 branch; this issue will close when that gets merged to devel.

**Comment by arunoda** [comment]
Thanks a lot.
